index.js

```diff
@@ -201,16 +211,20 @@ function parse(input, options) {
 		// Missing `=` should be `null`:
 		// http://w3.org/TR/2012/WD-url-20120524/#collect-url-parameters
 		value = value === undefined ? null : decode(value, options);
-
-		if (options.parseNumbers && !Number.isNaN(Number(value)) && (typeof value === 'string' && value.trim() !== '')) {
-			value = Number(value);
-		} else if (options.parseBooleans && value !== null && (value.toLowerCase() === 'true' || value.toLowerCase() === 'false')) {
-			value = value.toLowerCase() === 'true';
-		}
-
 		formatter(decode(key, options), value, ret);
 	}
 
+	Object.keys(ret).forEach(key => {
+		const value = ret[key];
+		if (Boolean(value) && typeof value === 'object') {
```

Comment:
Shouldn't this test for array?

Reply:
If `arrayFormat` is set to `index`, ret would be an object. I want to make this function generic so it can work both on array and object.

Reply:
`Boolean(value) &&` can be removed. And the check should be:

```suggestion
		if (typeof value === 'object' && value !== null) {
```